File: LongestSubsequenceWithLimitedSum.java
import java.util.Arrays;

class Solution {
    public int[] answerQueries(int[] nums, int[] queries) {
        int sol[] = new int[queries.length];
        Arrays.sort(nums);
        for(int i=0;i<sol.length;i++){
            int tmp = 0; sol[i] = nums.length;
            for(int j=0;j<nums.length;j++){
                tmp+=nums[j];
                if(tmp>queries[i]){
                    tmp-=nums[j];
                    sol[i]=j;
                    break;
                }
            }
        }
        return sol;
    }
}
public class LongestSubsequenceWithLimitedSum {
    public static void main(String[] args) {
        System.out.println("Hello world!");
    }
}

File: LongestSubsequenceWithLimitedSum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        sol = []
        nums.sort()
        for i in range(1,len(nums)):
            nums[i] += nums[i-1]

        for query in queries:
            index = bisect.bisect_right(nums, query)
            sol.append(index)

        return sol
            
